contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -10111,6 +10120,30 @@ bool reset_search_path(PLtsql_stmt_execsql *stmt, char *old_search_path, bool* r
 		else if(top_es_entry->estate && top_es_entry->estate->err_stmt &&
 				top_es_entry->estate->err_stmt->cmd_type == PLTSQL_STMT_EXEC_BATCH)
 			return false;
+
+		/* Traverse through the estate stack, if the stmt is inside trigger
+		 * we set the search path
+		 */
+		if(top_es_entry->estate && top_es_entry->estate->err_stmt &&
```

Comment:
I feel like it would be more straight forward if we check 
top_es_entry->estate && top_es_entry->estate->err_stmt  in the beginning of the loop and use a switch to redirect different branches. 
And if you see it is better to keep current form, would it be better if it is another "else if " instead of "if"?

Reply:
Moving ahead with the current form and replaced 'if' with `else if` to maintain the consistency. There were some cases which failed in switch so finding it better to have this approach.